### PR TITLE
chore: fix some errors detected by staticcheck

### PR DESCRIPTION
**Pull request by yin1999**
fix some errors detected by staticcheck

### CODE REVIEW

File: app/router/command/command_test.go [hunk 1]
```diff
@@ -80,7 +81,7 @@ func TestServiceSubscribeRoutingStats(t *testing.T) {
 	// Client goroutine
 	go func() {
 		defer lis.Close()
-		conn, err := grpc.DialContext(context.Background(), "bufnet", grpc.WithContextDialer(bufDialer), grpc.WithInsecure())
```
Comment (yin1999) [inline app/router/command/command_test.go:83]: `grpc.WithInsecure()` has been deprecated

File: common/drain/drainer.go [hunk 2]
```diff
@@ -36,7 +35,7 @@ func (d *BehaviorSeedLimitedDrainer) Drain(reader io.Reader) error {
 }
 
 func drainReadN(reader io.Reader, n int) error {
-	_, err := io.CopyN(ioutil.Discard, reader, int64(n))
```
Comment (yin1999) [inline common/drain/drainer.go:39]: `ioutil` package has been deprecated

File: proxy/socks/server.go [hunk 3]
```diff
@@ -101,7 +101,7 @@ func (s *Server) processTCP(ctx context.Context, conn stat.Connection, dispatche
 	reader := &buf.BufferedReader{Reader: buf.NewReader(conn)}
 	request, err := svrSession.Handshake(reader, conn)
 	if err != nil {
-		if inbound != nil && inbound.Source.IsValid() {
```
Comment (yin1999) [inline proxy/socks/server.go:104]: nil check has been done in line 90

File: common/dice/dice.go [hunk 4]
```diff
@@ -46,7 +45,3 @@ func (dd *DeterministicDice) Roll(n int) int {
 	}
 	return dd.Intn(n)
 }
-
-func init() {
-	rand.Seed(time.Now().Unix())
-}
```
Comment (yin1999) [inline common/dice/dice.go:52]: This is not necessary anymore, as [after go 1.20, the seed will be randomly generated](https://pkg.go.dev/math/rand#Seed). And the runtime `fastrand64` might be used for lock-free random number generating.

see also: https://cs.opensource.google/go/go/+/refs/tags/go1.22.0:src/math/rand/rand.go;l=308-312

**Comment by yuhan6665** [comment]
Looks great. Thanks for the writeup

**Pull request merged**